src/manifest.rs

```diff
@@ -75,7 +75,7 @@ pub type DependencyMap = HashMap<PackageName, DependencyManifest>;
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Manifest {
     /// Metadata about the root package
-    pub package: PackageManifest,
+    pub package: Option<PackageManifest>,
```

Comment:
Would it perhaps make sense to have a separate `PublishableManifest` type which has a non-optional `PackageManifest`? Something like:

```rust
impl Manifest
    fn publishable(&self) -> Option<PublishableManifest>;
}
```

And then all places where we need a manifest and expect it to have a `[package]` section, this is the type it will take. Not sure if this makes sense :)

Reply:
Well, you can only create a `Package` if the `PackageManifest` is found in the manifest